pkg/solana/txm/pendingtx_test.go

```diff
@@ -32,51 +30,818 @@ func TestPendingTxContext(t *testing.T) {
+		sig, cancel := newProcess()
+		msg := pendingTx{id: uuid.NewString()}
+		err := txs.New(msg, sig, cancel)
 		assert.NoError(t, err)
-		ids[sig] = id
+		ids[sig] = msg.id
 	}
 
 	// cannot add signature for non existent ID
-	require.Error(t, txs.Add(uuid.New(), solana.Signature{}))
+	require.Error(t, txs.AddSignature(uuid.New().String(), solana.Signature{}))
 
 	// return list of signatures
 	list := txs.ListAll()
 	assert.Equal(t, n, len(list))
 
 	// stop all sub processes
 	for i := 0; i < len(list); i++ {
-		id := txs.Remove(list[i])
+		id, err := txs.Remove(list[i])
+		assert.NoError(t, err)
 		assert.Equal(t, n-i-1, len(txs.ListAll()))
 		assert.Equal(t, ids[list[i]], id)
 
 		// second remove should not return valid id - already removed
-		assert.Equal(t, uuid.Nil, txs.Remove(list[i]))
+		id, err = txs.Remove(list[i])
+		require.Error(t, err)
+		assert.Equal(t, "", id)
 	}
 	wg.Wait()
 }
 
+func TestPendingTxContext_new(t *testing.T) {
+	t.Parallel()
+	_, cancel := context.WithCancel(tests.Context(t))
+	sig := randomSignature(t)
+	txs := newPendingTxContext()
+
+	// Create new transaction
+	msg := pendingTx{id: uuid.NewString()}
+	err := txs.New(msg, sig, cancel)
+	require.NoError(t, err)
+
+	// Check it exists in signature map
+	id, exists := txs.sigToID[sig]
+	require.True(t, exists)
+	require.Equal(t, msg.id, id)
+
+	// Check it exists in broadcasted map
+	tx, exists := txs.broadcastedTxs[msg.id]
+	require.True(t, exists)
+	require.Len(t, tx.signatures, 1)
+	require.Equal(t, sig, tx.signatures[0])
+
+	// Check status is Broadcasted
+	require.Equal(t, Broadcasted, tx.state)
+
+	// Check it does not exist in confirmed map
+	tx, exists = txs.confirmedTxs[msg.id]
+	require.False(t, exists)
+
+	// Check it does not exist in finalized map
+	tx, exists = txs.finalizedErroredTxs[msg.id]
+	require.False(t, exists)
+}
+
+func TestPendingTxContext_add_signature(t *testing.T) {
+	t.Parallel()
+	_, cancel := context.WithCancel(tests.Context(t))
+	sig1 := randomSignature(t)
+	sig2 := randomSignature(t)
+	txs := newPendingTxContext()
+
+	// Create new transaction
+	msg := pendingTx{id: uuid.NewString()}
+	err := txs.New(msg, sig1, cancel)
+	require.NoError(t, err)
+
+	err = txs.AddSignature(msg.id, sig2)
+	require.NoError(t, err)
+
+	// Check signature map
+	id, exists := txs.sigToID[sig1]
+	require.True(t, exists)
+	require.Equal(t, msg.id, id)
+	id, exists = txs.sigToID[sig2]
+	require.True(t, exists)
+	require.Equal(t, msg.id, id)
+
+	// Check broadcasted map
+	tx, exists := txs.broadcastedTxs[msg.id]
+	require.True(t, exists)
+	require.Len(t, tx.signatures, 2)
+	require.Equal(t, sig1, tx.signatures[0])
+	require.Equal(t, sig2, tx.signatures[1])
+
+	// Check confirmed map
+	tx, exists = txs.confirmedTxs[msg.id]
+	require.False(t, exists)
+
+	// Check finalized map
+	tx, exists = txs.finalizedErroredTxs[msg.id]
+	require.False(t, exists)
+}
+
+func TestPendingTxContext_on_broadcasted_processed(t *testing.T) {
+	t.Parallel()
```

Comment:
Should we include checks on:
- Processed to Finalized transitions as mentioned in [this comment](https://github.com/smartcontractkit/chainlink-solana/pull/909#discussion_r1829875519)?

Reply:
I actually have that test under the tests for the finalized map [here](https://github.com/smartcontractkit/chainlink-solana/pull/909/files#diff-ef0d3e62fe23dcbd48fbea8fd84e018d0bc97e17163e852f6e42ecda168caa4cR396)